src/AutoRest.CSharp/Input/CodeModelPartials.cs

```diff
@@ -90,6 +91,19 @@ public string[] Formats
         public bool SkipEncoding => TryGetValue("x-ms-skip-url-encoding", out var value) && Convert.ToBoolean(value);
     }
 
+    internal partial class OperationGroup
```

Comment:
Consider encapsulating this information on Mgmt build context instead of making it accessible to everyone but keeping it uninitialized.
We were also trying to keep the input model immutable.

Reply:
I don't think this makes it immutable as much as it adds decoration to it.

Reply:
It makes them mutable, you are adding stable properties and set their values.

Reply:
100% agreed, no mutability if at all possible.